draft-ietf-quic-qpack.md

```diff
@@ -101,19 +101,238 @@ balance between resilience against head-of-line blocking and optimal compression
 ratio.  The design goals are to closely approach the compression ratio of HPACK
 with substantially less head-of-line blocking under the same loss conditions.
 
-QPACK preserves the ordering of header fields within each header list.  An
-encoder MUST emit header field representations in the order they appear in the
-input header list.  A decoder MUST must emit header fields in the order their
-representations appear in the input header block.
+## Conventions and Definitions
 
-# Header Tables
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this
+document are to be interpreted as described in BCP 14 {{!RFC2119}} {{!RFC8174}}
+when, and only when, they appear in all capitals, as shown here.
+
+Definitions of terms that are used in this document:
+
+Header field:
+
+: A name-value pair sent as part of an HTTP message.
+
+Header list:
+
+: The ordered collection of header fields associated with an HTTP message.  A
+  header list can contain multiple header fields with the same name.  It can
+  also contain duplicate header fields.
+
+Header block:
+
+: The compressed representation of a header list.
+
+Encoder:
+
+: An implementation which transforms a header list into a header block.
+
+Decoder:
+
+: An implementation which transforms a header block into a header list.
+
+<!-- new definitions -->
+Absolute Index:
+
+: A unique index for each entry in the dynamic table.
+
+Base Index:
+
+: An absolute index in a header block from which relative indices are made
+
+QPACK is a name, not an acronym.
+
+## Notational Conventions
+
+Diagrams use the format described in Section 3.1 of {{?RFC2360}}, with the
+following additional conventions:
+
+x (A)
+: Indicates that x is A bits long
+
+x (A+)
+: Indicates that x uses the prefixed integer encoding defined in Section 5.1 of
+  [RFC7541], beginning with an A-bit prefix.
+
+x ...
+: Indicates that x is variable-length and extends to the end of the region.
+
+# Compression Process Overview
 
 Like HPACK, QPACK uses two tables for associating header fields to indices.  The
 static table (see {{table-static}}) is predefined and contains common header
 fields (some of them with an empty value).  The dynamic table (see
 {{table-dynamic}}) is built up over the course of the connection and can be used
 by the encoder to index header fields repeated in the encoded header lists.
 
+- The encoder uses a unidirectional stream to modify the state of the dynamic
```

Comment:
This bulleted list feels like it comes out of nowhere.  Perhaps this could be turned into something more narrative, or at least introduced?